virtual-desktop/src/app/window-manager/mvd-window-manager/shared/window-manager.service.ts

```diff
@@ -429,12 +432,32 @@ export class WindowManagerService implements MVDWindowManagement.WindowManagerSe
     };
   }
 
+  private savePluginData(plugin:ZLUX.Plugin,windowId:number,data:any){
+    let pathToSave : any = 'pluginData' + '/' + 'app'
+    let fileNameToSave : string = plugin.getIdentifier() + '-' + windowId
```

Comment:
windowId makes the filename unique, but that may become a problem on restoration. 
You will get a few files,
pluginA-1
pluginA-6
pluginA-8

When you go to restore, you restore them all, but now they are 0, 1, 2. So, then you may see
pluginA-0
pluginA-1
pluginA-2

That's also OK, but what would not be OK would be:
pluginA-0
pluginA-1
pluginA-2
pluginA-6
pluginA-8

Because then you went from 3 to 5, which is not what the user saw.

There could be 2 solutions:

1. After successful load of the autosave data, mark a boolean so that the first time after load that a new autosave will occur, it will first delete the old autosave data. You can use this.http.delete to delete content. In this way, you will prevent going from 3 to 5 in the above example.

2. Instead of saving 1 file per plugin instance to the config service, just save 1 file for all, where the file is an array of the data from the instances. In this way, you do not need to worry about cleanup of old files. It will always be replacing the old data with the new data.

In 2, you need to do a GET of the file at startup, and then all future autosave events just overwrite the file.
The file can look like
```
{ instances: 
[ 
  { data1: "abc", data2: false, data3: -3 },
  { data1: "def", data2: true, data3: -3 }
  { data1: "xzc", data2: false, data3: 0 }
]
}
```

In this way, it does not matter what the window Id was, and the order is also unimportant, because this file would be constantly overwritten. All that matters is that there were 3 instances of the plugin, not the order or the unique number.

Reply:
nevermind, it appears you already did option 1 which is fine.